demonstrations/tutorial_block_encoding.py

```diff
@@ -0,0 +1,413 @@
+r"""
+
+Block encoding with matrix access oracles
+=========================================
+
+
+Prominent quantum algorithms such as quantum phase estimation and quantum singular value
+transformation sometimes need to use **non-unitary** matrices inside quantum circuits. This is problematic
+because quantum computers can only perform unitary evolutions 🔥. Block encoding is a technique
+that solves this problem by embedding a non-unitary operator as a sub-block of a larger unitary 
+matrix 🧯.
+
+In previous demos we have discussed methods for `simulator-friendly <https://pennylane.ai/qml/demos/tutorial_intro_qsvt#transforming-matrices-encoded-in-matrices>`_
+encodings and block encodings using `linear combination of unitaries <https://pennylane.ai/qml/demos/tutorial_lcu_blockencoding>`_
+(LCU) decompositions. In this tutorial we explore another general block encoding framework that can be
+very efficient for sparse and structured matrices: block encoding with matrix access oracles.
+
+.. figure:: ../demonstrations/block_encoding/thumbnail_Block_Encodings_Matrix_Oracle.png
+    :align: center
+    :width: 50%
+    :target: javascript:void(0)
+
+A general circuit for block encoding an arbitrary matrix :math:`A \in \mathbb{C}^{N \times N}` with
+:math:`N = 2^{n}` can be constructed as shown in the figure below, if we have access to the oracles 
+:math:`U_A` and :math:`U_B`:
+
+.. figure:: ../demonstrations/block_encoding/general_circuit.png
+    :width: 50%
+    :align: center
+
+Where the :math:`H^{\otimes n}` operation is a Hadamard transformation on :math:`n` qubits. The
+:math:`U_A` operation is an oracle which encodes the matrix element :math:`A_{i,j}` into the the 
+amplitude of the ancilla qubit. The :math:`U_B` oracle ensures that we iterate over every 
+combination of :math:`(i,j)`.
+
+Finding an optimal quantum gate decomposition that implements :math:`U_A` and :math:`U_B` is not 
+always possible. We now explore two different approaches for constructing these oracles that can be 
+very efficient for matrices with specific structure or sparsity.
+
+Block encoding structured matricies 
+-----------------------------------
+In order to better understand the oracle access framework let us first define :math:`U_A` and :math:`U_B`
+For the exact block-encoding of :math:`A`. The :math:`U_A` oracle is responsible for encoding the 
```

Comment:
```suggestion
for the exact block-encoding of :math:`A`. The :math:`U_A` oracle is responsible for encoding the 
```

is this change correct, right?